extensions/blocks/revue/edit.js

```diff
@@ -0,0 +1,178 @@
+
+export default function RevueEdit( props ) {
+	const { attributes, className, setAttributes } = props;
+
+	useEffect( () => {
+		const validatedAttributes = getValidatedAttributes( defaultAttributes, attributes );
+		if ( ! isEqual( validatedAttributes, attributes ) ) {
+			setAttributes( validatedAttributes );
+		}
+	}, [ attributes ] );
+
+	const {
+		revueUsername,
+		emailLabel,
+		emailPlaceholder,
+		firstNameLabel,
+		firstNamePlaceholder,
+		firstNameShow,
+		lastNameLabel,
+		lastNamePlaceholder,
+		lastNameShow,
+	} = attributes;
+
+	const [ username, setUsername ] = useState( '' );
+
+	useEffect( () => {
+		if ( ! username && revueUsername ) {
+			setUsername( revueUsername );
+		}
+	}, [] );
+
+	const saveUsername = event => {
+		event.preventDefault();
+		setAttributes( { revueUsername: username } );
+	};
+
+	const supportLink =
+		isSimpleSite() || isAtomicSite()
+			? 'http://support.wordpress.com/wordpress-editor/blocks/revue-block/'
+			: 'https://jetpack.com/support/jetpack-blocks/revue-block/';
+
+	return (
+		<div className={ className }>
+			{ ! revueUsername && (
+				<Placeholder
+					icon={ <BlockIcon icon={ icon } /> }
+					instructions={ __( 'Enter your Revue username.', 'jetpack' ) }
+					label={ __( 'Revue', 'jetpack' ) }
+				>
+					<form onSubmit={ saveUsername }>
+						<input
+							className="components-placeholder__input"
+							onChange={ event => setUsername( event.target.value ) }
+							placeholder={ __( 'Enter your Revue username here…', 'jetpack' ) }
+							type="text"
+							value={ username }
+						/>
+						<div>
+							<Button disabled={ ! username } isDefault isLarge isSecondary type="submit">
+								{ __( 'Add Form', 'jetpack' ) }
+							</Button>
+						</div>
+					</form>
+					<div className={ `components-placeholder__learn-more` }>
+						<ExternalLink href={ supportLink }>
+							{ __( 'Need help finding your Revue username?', 'jetpack' ) }
+						</ExternalLink>
+					</div>
+				</Placeholder>
+			) }
+
+			{ revueUsername && (
+				<>
+					<InspectorControls>
+						<PanelBody title={ __( 'Form Settings', 'jetpack' ) }>
+							<ToggleControl
+								label={ __( 'Show first name field.', 'jetpack' ) }
+								checked={ firstNameShow }
+								onChange={ () => setAttributes( { firstNameShow: ! firstNameShow } ) }
+							/>
+							<ToggleControl
+								label={ __( 'Show last name field.', 'jetpack' ) }
+								checked={ lastNameShow }
+								onChange={ () => setAttributes( { lastNameShow: ! lastNameShow } ) }
+							/>
+						</PanelBody>
+					</InspectorControls>
+
+					<BlockControls>
+						<Toolbar>
+							<IconButton
+								className="components-toolbar__control"
+								label={ __( 'Edit Username', 'jetpack' ) }
+								icon="edit"
+								onClick={ () => setAttributes( { revueUsername: undefined } ) }
+							/>
+						</Toolbar>
+					</BlockControls>
+
+					<TextControl
+						label={
+							<JetpackFieldLabel
+								label={ emailLabel }
```

Comment:
A very minor thing, but in the edit view of the label field the text is not vertically centred, but in the other form fields it is:
<img width="177" alt="Screen Shot 2020-02-25 at 9 52 15 AM" src="https://user-images.githubusercontent.com/3629020/75190218-b32caa00-57b4-11ea-8852-0783d2b0477a.png">